Makefile

```diff
@@ -136,7 +136,7 @@ check:
 	! golint $(PKG) | grep -vE '(\.pb\.go|embedded\.go|yyEofCode|_string\.go|LastInsertId)'
 	! gofmt -s -l . | grep -vF 'No Exceptions'
 	! goimports -l . | grep -vF 'No Exceptions'
-	go vet $(PKG)
+	! go tool vet --shadow --shadowstrict . 2>&1 | grep -vE '(\.pb\.go|declaration of err shadows)'
```

Comment:
On Thu, Jul 9, 2015 at 11:09 AM, Ben Darnell <notifications@github.com>
wrote:

> In Makefile
> <https://github.com/cockroachdb/cockroach/pull/1649#discussion_r34265441>:
>
> > @@ -136,7 +136,7 @@ check:
> >  	! golint $(PKG) | grep -vE '(\.pb\.go|embedded\.go|yyEofCode|_string\.go|LastInsertId)'
> >  	! gofmt -s -l . | grep -vF 'No Exceptions'
> >  	! goimports -l . | grep -vF 'No Exceptions'
> > -	go vet $(PKG)
> > +	! go tool vet --shadow --shadowstrict . 2>&1 | grep -vE '(\.pb\.go|declaration of err shadows)'
>
> go vet is weird. Last time I checked, some checks do not work when it is
> invoked as go tool vet (but this is also the only way to pass args like
> --shadow. Do we need to run it both ways?).
>

go vet just runs "go tool vet" without options, which means --all=true (the
default). However, --shadow is experimental and not enabled by --all. So
the change really is the same as going from "go tool vet" to "go tool vet
--shadow"


>  The main feature of go-nyet is shadow checking, but it clearly didn't
> catch everything that go vet did. Can/should we get rid of go-nyet if
> we're going to start using vet --shadow?
>
We probably should remove go-nyet. It wasn't catching any of the shadowing
reported by go vet, and it has its own set of problems (see exception
pattern in the makefile).


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1649/files#r34265441>.
>